pkg/distributor/validate_test.go

```diff
@@ -236,10 +262,34 @@ func TestValidateLabels(t *testing.T) {
 			metric:                   map[model.LabelName]model.LabelValue{model.MetricNameLabel: "foo", "label1": "abc\xfe\xfddef"},
 			skipLabelNameValidation:  true,
 			skipLabelCountValidation: false,
+			config:                   legacyConfig,
 			err:                      nil,
 		},
+		{
+			metric:                   map[model.LabelName]model.LabelValue{model.MetricNameLabel: "foo", "name😀": "value", "team": "b"},
+			skipLabelNameValidation:  false,
+			skipLabelCountValidation: false,
+			config:                   legacyConfig,
+			err: fmt.Errorf(
+				invalidLabelMsgFormat,
+				"name😀",
+				mimirpb.FromLabelAdaptersToString(
+					[]mimirpb.LabelAdapter{
+						{Name: model.MetricNameLabel, Value: "foo"},
+						{Name: "name😀", Value: "value"},
+						{Name: "team", Value: "b"},
+					},
+				),
+			),
+		},
+		{
+			metric:                   map[model.LabelName]model.LabelValue{model.MetricNameLabel: "foo", "name😀": "value", "team": "b"},
+			skipLabelNameValidation:  false,
+			skipLabelCountValidation: false,
+			config:                   utf8Config,
+		},
 	} {
-		err := validateLabels(s, cfg, userID, "custom label", mimirpb.FromMetricsToLabelAdapters(c.metric), c.skipLabelNameValidation, c.skipLabelCountValidation, cast, ts)
+		err := validateLabels(s, c.config, userID, "custom label", mimirpb.FromMetricsToLabelAdapters(c.metric), c.skipLabelNameValidation, c.skipLabelCountValidation, cast, ts)
```

Comment:
i'm tempted to add more test cases for utf8 config. I think a lot of the existing cases would work in both validation schemes. Maybe have a slice of schemes to run each test cases in? or have two expected errors - one for legacy validation scheme and one for utf8